diff --git a/exclude.cpp b/exclude.cpp
index 8ce138a..d5b2730 100644
--- a/exclude.cpp
+++ b/exclude.cpp
@@ -38,6 +38,7 @@ TWExclude::TWExclude() {
 	add_relative_dir(".");
 	add_relative_dir("..");
 	add_relative_dir("lost+found");
+	add_relative_dir("sparse_image");
 }
 
 void TWExclude::add_relative_dir(const string& dir) {
@@ -81,7 +82,7 @@ uint64_t TWExclude::Get_Folder_Size(const string& Path) {
 		}
 		if ((st.st_mode & S_IFDIR) && !check_skip_dirs(FullPath) && de->d_type != DT_SOCK) {
 			dusize += Get_Folder_Size(FullPath);
-		} else if (st.st_mode & S_IFREG || st.st_mode & S_IFLNK) {
+		} else if ((st.st_mode & S_IFREG || st.st_mode & S_IFLNK) && !check_skip_dirs(FullPath)) {
 			dusize += (uint64_t)(st.st_size);
 		}
 	}
